### PR TITLE
feat(cli): add two-tier RPG data management (CI + local)

### PR DESCRIPTION
**Pull request by amondnet**
## Summary

This PR implements a two-tier RPG data management architecture with CI-based canonical graph updates and local branch-specific evolution.

## Two-Tier Architecture

**Tier 1 (CI - Canonical Graph):**
- On main push, GitHub Actions runs `rpg encode` or `rpg evolve`
- Commits `.rpg/graph.json` to git as the canonical RPG
- Uses `--stamp` flag to track the last encoded commit SHA in `config.github.commit`

**Tier 2 (Local - Branch Evolution):**
- `rpg sync` copies canonical graph to `.rpg/local/`
- Applies incremental evolve for local branch changes
- Builds vector indices for semantic search
- All local data is gitignored

**Directory Structure:**
```
.rpg/
  graph.json          # Canonical RPG (git committed, CI-managed)
  config.json         # Encode/sync settings (git committed)
  local/              # Local-only data (gitignored)
    graph.json        # Local evolved RPG copy
    vectors/          # LanceDB vector embeddings
    state.json        # Local state (base commit, branch, etc.)
```

## New CLI Commands

1. **`rpg init [path]`** - Initialize RPG structure in repository
   - `--hooks` - Install git pre-push hook
   - `--ci` - Generate GitHub Actions workflow
   - `--encode` - Run initial encode

2. **`rpg sync`** - Sync canonical graph to local with incremental evolve
   - `--force` - Force re-sync from scratch

3. **`rpg stamp <file>`** - Stamp config.github.commit with current HEAD SHA

4. **`rpg last-commit <file>`** - Print last encoded commit SHA

5. **Enhanced `rpg encode/evolve`** - Added `--stamp` option to record commit

## Changes

**Modified:**
- `.gitignore` - Added `.rpg/local/` pattern
- `CLAUDE.md` - Documented two-tier architecture and setup workflow
- `packages/cli/src/cli.ts` - Added `--stamp` option to encode/evolve commands, registered init/sync subcommands, added stamp/last-commit commands
- `packages/graph/src/rpg.ts` - Added `updateConfig()` method for mutable config updates
- `packages/utils/package.json` - Added `./git-helpers` subpath export

**New Files:**
- `packages/utils/src/git-helpers.ts` - Git utility functions (getHeadCommitSha, getCommitRange, etc.)
- `packages/cli/src/commands/init.ts` - `rpg init` command implementation
- `packages/cli/src/commands/sync.ts` - `rpg sync` command implementation
- `packages/cli/src/commands/hooks.ts` - Git hooks installer
- `packages/cli/src/templates/rpg-encode.yml` - GitHub Actions workflow template

**Tests:**
- `tests/rpg-config.test.ts` - Unit tests for updateConfig() method
- `tests/git-helpers.test.ts` - Unit tests for git helper functions
- `tests/cli-stamp.test.ts` - Unit tests for stamp/last-commit logic
- `tests/cli-init-sync.integration.test.ts` - Integration tests for init/sync commands

## Test Results

✅ **Unit Tests:** 34 test files / 603 tests passed
✅ **Integration Tests:** 1 test file / 6 tests passed

All tests verify:
- Commit tracking and stamping
- Git helper functions
- Config update behavior
- Init/sync workflow
- GitHub Actions workflow generation

## Usage Example

```bash
# Initialize with CI workflow and git hooks
rpg init --ci --hooks --encode

# Or step by step:
rpg init                     # Create .rpg/ structure
rpg encode . -o .rpg/graph.json --stamp  # Initial encode
rpg sync                     # Copy to local + evolve

# On feature branch:
git checkout -b feat/new-feature
# ... make changes ...
rpg sync                     # Evolve local graph with branch changes
```

## Breaking Changes

None - this is a new feature addition with backward compatibility.

## Related Issues

Implements the two-tier data management architecture discussed in repository planning.

---
## Summary by cubic
Implements two-tier RPG data management with a CI-stamped canonical graph and local branch evolution via rpg sync. Adds CI resilience with evolve→encode fallback and expands test coverage for sync, init, and git helpers.

- **New Features**
  - Two-tier flow: CI commits .rpg/graph.json on main; rpg sync manages .rpg/local/graph.json with branch changes and vectors.
  - CLI: rpg init (hooks, CI, optional encode), rpg sync (--force), rpg stamp, rpg last-commit; encode/evolve support --stamp.
  - CI: rpg-encode.yml runs encode/evolve using last-commit and falls back to full encode if evolve fails.
  - Config/Utils: updateConfig() for mutable config; git-helpers for HEAD SHA, merge-base, branch detection; shared HEAD-stamp helper.

- **Bug Fixes**
  - Init/CI: fixed workflow template path in init --ci; clearer init error hints.
  - Sync: improved evolve error logs; no duplicate canonical→local copy on default branch; correctly re-syncs when local is stale vs canonical; removed redundant needsCopy guard (CodeQL).
  - Git/hooks: getCurrentBranch re-throws non-detached-HEAD errors; getDefaultBranch adds a guard and warns on hardcoded fallback; hooks installer ensures .git/hooks exists and prints actionable messages.
  - Testing: exported ensureGitignoreEntry/generateCIWorkflow; replaced simulated tests with Commander-based runs; added sync and hooks edge-case tests; set default branch to main in tests; expanded coverage for corrupt local state, baseCommit re-copy, default-branch no-op, feature-branch evolve, detached HEAD, and default-branch fallback.

<sup>Written for commit 9f7ff19acc6da7c4eb9894037f6f2a71bbdeae63. Summary will update on new commits.</sup>

### CODE REVIEW

File: packages/cli/src/commands/sync.ts [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+          }
+          catch {
+            log.warn('Could not parse local state, will rebuild')
+          }
+        }
+
+        const isOnDefaultBranch = currentBranch === defaultBranch || currentBranch === ''
+        const needsEvolve = !isOnDefaultBranch && canonicalCommit
+
+        if (options.force || !existsSync(localGraphPath) || !localState) {
```
[the remark this answers is not in the document]
Comment (amondnet) [reply packages/cli/src/commands/sync.ts]: ✅ Applied in 57849a9 — added `localState.baseCommit !== canonicalCommit` check to the `needsCopy` condition. Now when CI updates the canonical graph, `rpg sync` correctly detects the stale local copy and refreshes before evolving. Good catch!
